### PR TITLE
Add reactive health indicator for Cassandra

### PR DESCRIPTION
**Pull request by ayudovin**
Adding functionality for supporting actuator with reactive cassandra
this [enhancement](https://github.com/spring-projects/spring-boot/issues/11767)

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	 */
+	public CassandraReactiveHealthIndicator(ReactiveCassandraOperations reactiveCassandraOperations) {
+		Assert.notNull(reactiveCassandraOperations, "ReactiveCassandraOperations must not be null");
+		this.reactiveCassandraOperations = reactiveCassandraOperations;
+	}
+
+	@Override
+	protected Mono<Health> doHealthCheck(Health.Builder builder) {
+		Select select = QueryBuilder.select("release_version").from("system", "local");
+		Mono<ReactiveResultSet> results = this.reactiveCassandraOperations.getReactiveCqlOperations()
```
Comment (mp911de) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java]: How about simplifying the CQL call to:

```java
Mono<String> version = template.getReactiveCqlOperations().queryForObject(select, String.class);
```
?

This requires less mocking in the tests. Would be also good to improve `CassandraHealthIndicator` to call `queryForObject(…)`.

Comment (ayudovin) [reply spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java]: ok, I agree with you and have changed it, thx

**Comment by snicoll** [comment]
@ayudovin thank you very much for your first contribution to Spring Boot. This is now merged in `master` with a [polish commit](https://github.com/spring-projects/spring-boot/commit/af0aa11d15e5bcb98c57458cb3592b4696345cc1).

**Comment by ayudovin** [comment]
Thank you

**Pull request closed without merging**